implementations/contracts/ERC725XCore.sol

```diff
@@ -49,6 +49,29 @@ abstract contract ERC725XCore is OwnableUnset, ERC165, IERC725X {
         return _execute(operationType, to, value, data);
     }
 
+    /**
+     * @inheritdoc IERC725X
+     */
+    function execute(
+        uint256[] memory operationsType,
+        address[] memory to,
+        uint256[] memory values,
+        bytes[] memory data
+    ) public payable virtual onlyOwner returns (bytes[] memory result) {
+        if (
+            operationsType.length != to.length ||
+            (to.length != values.length || values.length != data.length)
+        ) revert ERC725X_ExecuteParametersLengthMismatch();
+
+        result = new bytes[](operationsType.length);
+        for (uint256 i = 0; i < operationsType.length; i++) {
+            if (address(this).balance < values[i])
```

Comment:
here shouldnt it be `>=` to the sum of all values?

Reply:
For example I have a balance of 2eth, if I have 3 operations of 1eth. It would not revert?

Reply:
nvm. My comment is erronated, balance will be updated on each loop